packages/create-react-app/index.js

```diff
@@ -182,7 +182,29 @@ function run(root, appName, version, verbose, originalDirectory, template) {
 
   install(allDependencies, verbose, function(code, command, args) {
     if (code !== 0) {
-      console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
+      console.log();
+      console.error('Aborting installation.', chalk.cyan(command + ' ' + args.join(' ')), 'has failed.');
+      // On 'exit' we will delete these files from target directory.
+      var knownGeneratedFiles = [
+        'package.json', 'npm-debug.log', 'yarn-error.log', 'yarn-debug.log', 'node_modules'
+      ];
+      var currentFiles = fs.readdirSync(path.join(root));
+      currentFiles.forEach(function (file) {
+        knownGeneratedFiles.forEach(function (fileToMatch) {
+          // This will catch `*(npm-debug|yarn-error|yarn-debug).log*` files.
+          if ((fileToMatch.indexOf('.log') !== -1 && file.indexOf(fileToMatch) !== -1) || file === fileToMatch) {
```

Comment:
We might want this to be `file.indexOf(fileToMatch) === 0` so that it only globs the end of the file.

Anyway, I've got to head to bed. Goodnight from the east coast!